docs/docs/references/cli/exec/index.md

```diff
@@ -0,0 +1,166 @@
+
+- `-p`, `--publisher`
+	- Description: Where to publish the result of the job.
+
+- `-i`, `--input`
+    - Description: Mount URIs as inputs to the job. Can be specified multiple times. Format: src=URI,dst=PATH[,opt=key=value]
+        ### Examples:
+        **Mount IPFS CID to /inputs directory**
+
+        `-i ipfs://QmeZRGhe4PmjctYVSVHuEiA9oSXnqmYa4kQubSHgWbjv72`
+
+        **Mount S3 object to a specific path**
+
+        `-i s3://bucket/key,dst=/my/input/path`
+
+        **Mount S3 object with specific endpoint and region**
+
+        `-i src=s3://bucket/key,dst=/my/input/path,opt=endpoint=https://s3.example.com,opt=region=us-east-1`
+
+- `-o`, `--output`
+    - Description: name:path of the output data volumes. 'outputs:/outputs' is always added unless '/outputs' is mapped to a different name.
+
+- `-e`, `--env`
+    - Description: The environment variables to supply to the job (e.g. --env FOO=bar --env BAR=baz)
+
+- `--timeout`
+    - Description:  Job execution timeout in seconds (e.g. 300 for 5 minutes)
+
+- `-l`, `--labels`
+    - Description: List of labels for the job. Enter multiple in the format '-l a -l 2'. All characters not matching /a-zA-Z0-9_:|-/ and all emojis will be stripped.
+
+- `-s`, `--selector`
+    - Description: Selector (label query) to filter nodes on which this job can be executed, supports '=', '==', and '!='.(e.g. -s key1=value1,key2=value2). Matching objects must satisfy all of the specified label constraints.
+
+
+## Global Flags
+
+- `--api-host string`:
+    - Description: Specifies the host used for RESTful communication between the client and server. The flag is disregarded if the `BACALHAU_API_HOST` environment variable is set.
+    - Default: `bootstrap.production.bacalhau.org`
+
+- `--api-port int`:
+    - Description: Specifies the port for REST communication. If the `BACALHAU_API_PORT` environment variable is set, this flag will be ignored.
+    - Default: `1234`
+
+- `--log-mode logging-mode`:
+    - Description: Sets the desired log format. Options are: `default`, `station`, `json`, `combined`, and `event`.
+    - Default: `default`
+
+- `--repo string`:
+    - Description: Defines the path to the bacalhau repository.
+    - Default: ``$HOME/.bacalhau`
+
+
+## Examples
+
+### Running python tasks
+
+1. **Basic Usage**:
+
+   **Command**:
+   ```shell
+   → bacalhau exec python -- -c "import this"
+   ```
+
+   **Output**:
+   ```text
+   The Zen of Python, by Tim Peters
+
+   Beautiful is better than ugly.
+   Explicit is better than implicit.
+   Simple is better than complex.
+   Complex is better than complicated.
+   Flat is better than nested.
+   ....
+   ```
+
+2. **Single file Python**:
+
+   **Command**:
+   ```shell
+   → bacalhau exec --code=app.py python app.py
+   ```
+
+   where app.py is 
+   ```python
+   """
+   pip install colorama
+   """
+
+   from colorama import Fore
+   print(Fore.RED + "Hello World")
```

Comment:
not sure about color as an example - just do Fibonacci?

Reply:
Can do, but the point of this example was really to show the auto-requirements for single file scripts.  So maybe I can find something that does both.